modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -257,6 +258,49 @@ def preprocess_func(cls, func):
         """
         return cls.execution_wrapper.put(func)
 
+    @classmethod
+    def get_lengths_glob(cls, parts):
+        """
+        Get lengths of all the given parts.
+
+        Parameters
+        ----------
+        parts : list
+            All parts whose lengths are to be retrived.
+
+        Returns
+        -------
+        str
+        """
+        lengths = [part.length() for part in parts]
+        filter_condition = cls._engine_wrapper.check_is_future
+
+        apply_function_on_selected_items(
+            lengths, filter_condition, cls._engine_wrapper.materialize_func
+        )
+        return lengths
+
+    @classmethod
+    def get_widths_glob(cls, parts):
+        """
+        Get widths of all the given parts.
+
+        Parameters
+        ----------
+        parts : list
+            All parts whose widths are to be retrived.
+
+        Returns
+        -------
+        str
+        """
+        widths = [part.width() for part in parts]
+        filter_condition = cls._engine_wrapper.check_is_future
+        apply_function_on_selected_items(
+            widths, filter_condition, cls._engine_wrapper.materialize_func
+        )
+        return widths
+
```

Comment:
This code seems redundant. The `length()` and `width()` functions do the materialization by default and save the result.

Reply:
The length() and width() does materialization on each partition when these are called with materialization this becomes a ray[ anti pattern](https://docs.ray.io/en/latest/ray-core/patterns/ray-get-loop.html) ,  the idea here is to call length() and width without materialization and do the materialization in a list later.

Reply:
The `materialize` parameter is `True` by default for these methods. The code `[part.width() for part in parts]` creates a list of ints.

Reply:
Got it , Yes the parameter should be be false.